csrc/scheduler/pointwise.cpp

```diff
@@ -434,11 +434,11 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams& params) {
   // fusion segmentation
   scheduler_utils::clearMemorySpace(fusion);
 
-  // maybe has_reduction for scheduling should be done on a per output tensor
-  // basis.
-  TORCH_INTERNAL_ASSERT(
-      ir_utils::getReductionOps(fusion).empty(),
-      "This scheduler only handles pointwise ops.");
+  // Replace each reduction with a call to full(). Assumes we have already
+  // verified that this is safe to do.
+  for (auto rop : ir_utils::getReductionOps(fusion)) {
+    ir_utils::replaceReductionWithFull(rop);
```

Comment:
@naoyam , as we discussed this replaces reductions with full (after verifying that this is safe during `canScheduleRuntime`). The problem is that this replacement breaks the path from outputs to inputs. In a simple linear kernel such as in the test, `schedulePointwise` fails to find a reference tensor. In this kind of case, we actually have an unused input after the transformation, but I don't think there's a convenient way to eliminate that upstream branch from the replaced expression. Perhaps this could be done by introducing a new OptOutMutator that would remove upstream branches (if this doesn't already exist), since this would clear the uses() for any inputs that would have only dead branches, so the reference tensor would be cleared.

Reply:
Sounds strange to me that an unused input would the reference detection fail. Unused inputs would happen without this translation, and I think they would be just ignored, no?

Reply:
Actually, I think I can just use `ir_utils::replaceValue` for this.

Reply:
It becomes unused after we break the fusion at the reduction. However, the `uses()` of the input is still non-empty because that upstream branch is not cleared. Same happens if I use `ir_utils::replaceValue`.

Reply:
`OptOutMutator` only does the replacements/removals we register, so we need to do another pass to remove the upstream branch.

Reply:
This is no longer an issue, right?

Reply:
Should be.